DOC: Complete API for model_blender (#10512)
spacetelescope/jwst: docs/jwst/model_blender/main.rst

Reviewer: ```suggestion
This produces a ``product`` identical to a call to :func:`~jwst.model_blender.blendmeta.blendmodels` described above.
```

.. _blender_handbook:

Role of Model Blender
=====================

Model Blender performs 2 operations:

> - Accumulate, blend and set metadata when combining several models (

Reviewer: ```suggestion
- Accumulate, blend, and set metadata when combining several models (
```

have had different values for each attribute.  The ``model_blender`` package solves
this problem by allowing the user to define rules that can be used to determine a
final single value from all the input values for each attribute, using separate
rules for each attribute as appropriate.  This package also creates a
table that records the input attribute values for all the input models used to
create the final product, allowing the user to select what attributes to keep in
this table.

This code works by:

- reading in all the input datamodels (either already in-memory or from FITS files)
- evaluating the rules for each attribute as defined in the model's schema
- determining from definitions in the input model's schema what attributes to keep in the table
- applying each attributes rule to the set of input values to determine the final output value
- updating the output model's metadata with the new values
- generating the output table with one row for each input model's values

Using model_blender
===================

Using blendmodels
-----------------

The simplest way to run model blender only requires calling a single interface

> that is :func:`~jwst.model_blender.blendmeta.blendmodels`:

Reviewer: ```suggestion
The simplest way to run model blender only requires calling a single interface,
:func:`~jwst.model_blender.blendmeta.blendmodels`:
```
